java/ql/src/Security/CWE/CWE-90/LdapInjection.qhelp

```diff
@@ -0,0 +1,63 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>If an LDAP query is built using string concatenation, and the
+components of the concatenation include user input, a user 
+is likely to be able to run malicious LDAP queries.</p>
+</overview>
+
+<recommendation>
+<p>If user input must be included in an LDAP query, it should be escaped to
+avoid a malicious user providing special characters that change the meaning
+of the query. If possible build the LDAP query using framework helper methods, for example
+from Spring's <code>LdapQueryBuilder</code> and <code>LdapNameBuilder</code>,
+instead of string concatenation. Alternatively, escape user input using an appropriate
+LDAP encoding method, for example: <code>encodeForLDAP</code> or </code>encodeForDN</code>
```

Comment:
The `qhelp` preview test revealed one typo in a `code` tag. This should fix it.

```suggestion
LDAP encoding method, for example: <code>encodeForLDAP</code> or <code>encodeForDN</code>
```

Reply:
Thanks. I applied the fix.